blocks/blogheader/blogheader.js

```diff
@@ -0,0 +1,25 @@
+import { getMetadata, decorateIcons } from '../../scripts/aem.js';
+
+export default async function decorate(block) {
+  const blogHeaderMeta = getMetadata('blogheader');
+  const blogHeaderPath = blogHeaderMeta
+    ? new URL(blogHeaderMeta).pathname
+    : '/blogs/blog-nav';
+
+  const blogHeaderResp = await fetch(`${blogHeaderPath}.plain.html`);
```

Comment:
I don't have any concern with this implementation, it is a correct one, so next I'm just sharing for more information:
There is in the official Edge Delivery Block Collection library a `Fragment` block: https://www.aem.live/developer/block-collection/fragment which would be a more generalised implementation for this pattern.